src/common/transformations/src/transformations/mlir/mlir_op.cpp

```diff
@@ -332,10 +487,23 @@ NodePtr MLIROp::clone_with_new_inputs(const ov::OutputVector& new_args) const {
     return std::make_shared<MLIROp>(new_args, engine, output_types, dimensions_map);
 }
 
-bool MLIROp::evaluate(ov::TensorVector& outputs, const ov::TensorVector& inputs) const {
+bool MLIROp::evaluate(ov::TensorVector& outputs, const ov::TensorVector& inputs, const ov::EvaluationContext& evaluationContext) const {
+    if (!engine->requires_packed_args()) {
+        std::vector<void*> args;
```

Comment:
Why not just pass the inputs as is, without making a redundant memory allocation and copy?

Reply:
> where `MLIROp` is only responsible for inputs "preprocessing"

But you still doing the inputs processing in invoke(), but half of the processing is also done in evaluate(). IMHO it's redundant.

Reply:
Also, this branch is only compatible with MLIREvaluateGcGPU.

Reply:
done, now it passes tensor vectors as is to `::invoke()`